lib/Parse/ParseDecl.cpp

```diff
@@ -6822,6 +6843,85 @@ void Parser::ParseCheckedPointerSpecifiers(DeclSpec &DS) {
         Diag(StartLoc, DiagID) << PrevSpec;
 }
 
+void Parser::ParseForanySpecifier(DeclSpec &DS) {
+  EnterScope(Scope::DeclScope | Scope::ForanyScope);
+  SourceLocation ForAnyStartLoc = ConsumeToken();
+  if (ExpectAndConsume(tok::l_paren, diag::err_forany_unexpected_token)) {
+    ExitScope();
+    return;
+  }
+  tok::TokenKind prevToken = tok::l_paren;
+  // Loop through inside of (), to retrieve polymorphic type declarations.
+  bool breakOutOfWhileLoop = false;
+  while (!breakOutOfWhileLoop) {
+    // In here, it should be one of the three. tok::identifier, tok::comma, 
+    // tok::r_paren. Anything else will be an error.
+    if (Tok.getKind() == tok::identifier) {
+      // We'd expect the previous token to not be an identifier.
+      if (prevToken == tok::identifier) {
+        Diag(Tok.getLocation(), diag::err_forany_comma_or_parenthesis_expected);
+        SkipUntil(tok::r_paren, SkipUntilFlags::StopAtSemi);
+        breakOutOfWhileLoop = true;
+        break;
+      }
+
+      // Add polymorphic type to scope.
+      SourceLocation StartLoc = Tok.getLocation();
+      DeclSpec polyDS(AttrFactory);
+      PrintingPolicy Policy = Actions.getPrintingPolicy();
+      const char *PrevSpec = nullptr;
+      unsigned DiagID;
+      polyDS.SetStorageClassSpec(Actions, DeclSpec::SCS_typedef, StartLoc, 
+                                 PrevSpec, DiagID, Policy);
+      polyDS.SetTypeSpecType(DeclSpec::TST_void, StartLoc, PrevSpec, DiagID, 
+                             Policy);
+      DS.SetRangeStart(ForAnyStartLoc);
+      DS.SetRangeEnd(SourceLocation());
+      Declarator D(polyDS, Declarator::TheContext::BlockContext);
+      D.SetRangeBegin(ForAnyStartLoc);
+      D.SetRangeEnd(SourceLocation());
+      D.AddTypeInfo(DeclaratorChunk::getPointer(DS.getTypeQualifiers(), StartLoc,
```

Comment:
You are right. Thank you for catching this. This is completely wrong.This line should pick up information from the newly created decl spec, instead of the old decl spec.